aopy/postproc.py

```diff
@@ -136,6 +136,109 @@ def get_trial_targets(trials, targets):
+         targetpospertrial (ntrials, 2): X and Y pos of target
+
+    Returns:
+        (ntime, ntrials): Array including instantaneous direction to the target from the cursor [rad]
+
+    '''
+    ntime = trialaligned_xpos.shape[0]
+    ntrials = trialaligned_xpos.shape[1]
+    inst_target_dir = np.zeros((ntime, ntrials))*np.nan
+
+    for itrial in range(ntrials):
+        for itime in range(ntime):
+            cursor_location = [trialaligned_xpos[itime, itrial], trialaligned_ypos[itime, itrial]]
+            target_location = targetpospertrial[itrial, :]
+            rel_target_angle, _ = get_target_dir_from_cursor(cursor_location, target_location)
+            inst_target_dir[itime, itrial] = rel_target_angle
+
+    return inst_target_dir
+
+def mean_fr_inst_dir(data, cursorxpos, cursorypos, targetpos, data_binwidth, targetloc_binwidth, data_samplerate, cursor_samplerate):
+    '''
+    This function takes trial aligned neural data, cursor position, and target position and calculates
+    the mean firing rate per target location. 
+
+    Args:
+        data (ntime, nunit, ntrial): Trial aligned data
+        cursorxpos (ntime, ntrial): Trial aligned cursor position
+        cursorypos (ntime, ntrial): Trial aligned cursor position
+        targetpos (ntrial, 2): Target position for each trial. First column is x target position, second column is y target position.
+        data_binwidth (float): Bin size for neural data and cursor position. Can not be smaller than allowed by cursor position sampling rate.
+        targetloc_binwidth (float): Bin size for lumping target positions [deg]
+        data_samplerate (int): Sampling rate for data
+        cursor_samplerate (int): Sampling rate for cursor position
+
+    Returns:
+        (nunit, ndirection): Average firing rate per unit per direction bin.
+        
+    '''
+    # Check that binwidths are not lower than allowed by sampling rate
+    max_cursor_binwidth = 1/cursor_samplerate #[s/sample]
+    if data_binwidth < max_cursor_binwidth:
+        data_binwidth = max_cursor_binwidth
+    
+    ndatatime, nunit, ntrial = data.shape
+    ndirection = 360//targetloc_binwidth
+    nbins = math.ceil(ndatatime/(data_samplerate*data_binwidth)) # the number of bins
+
+    # Check target bin width is evenly spaced around task circle
+    if 360%targetloc_binwidth != 0:
+        warnings.warn("Target location bins are not evenly spaced. Please choose a bin width that evenly divides into 360 degrees")
+
+    # Bin neural data and cursor pos for each trial to make them the same samplingrate
+    binned_data = np.zeros((nbins, nunit, ntrial))*np.nan
+    for itrial in range(ntrial):
+        binned_data[:,:,itrial] = precondition.bin_spikes(data[:,:,itrial], data_samplerate, data_binwidth)
+
+    binned_cursorxpos = precondition.bin_spikes(cursorxpos, cursor_samplerate, data_binwidth)
+    binned_cursorypos = precondition.bin_spikes(cursorypos, cursor_samplerate, data_binwidth)
```

Comment:
hmmmmmm

i guess we should rename bin_spikes to something else, huh

Reply:
actually are you sure this works? seems to me summing cursor positions together is a bad idea

Reply:
My understanding was that it was averaging the cursor position for each bin

Reply:
i guess so? its summing the cursor positions then dividing by the bin width, so won't it be off by some factor of time to the arithmetic average cursor position? but i suppose that doesn't matter since the relative direction will be the same between any two points.

Reply:
oh i see, yes, let me think about this some more.